packages/frame-host/src/endpoint.ts

```diff
@@ -0,0 +1,81 @@
+import * as Comlink from "comlink";
+import { RefObject } from "react";
+import WebView, { WebViewMessageEvent } from "react-native-webview";
+
+export const SYMBOL_IGNORING_RPC_RESPONSE_ERROR: symbol = Symbol();
+
+export type WebViewEndpoint = Comlink.Endpoint & {
+  /**
+   * Manually distribute events to listeners as an alternative to `document.addEventHandler` which is unavailable in React Native.
+   */
+  onMessage: (e: WebViewMessageEvent) => void;
+};
+
+/**
+ * An endpoint of communicating with WebView
+ */
+export function createWebViewRpcEndpoint(ref: RefObject<WebView>): WebViewEndpoint & { emit: (data: any) => void } {
+  const listeners: EventListenerOrEventListenerObject[] = [];
+  return {
+    addEventListener: (type, listener) => {
+      if (type !== "message") {
+        throw Error(`Got an unexpected event type "${type}". Expected "message".`);
+      }
+      listeners.push(listener);
+    },
+    removeEventListener: (type, listener) => {
+      if (type !== "message") {
+        throw Error(`Got an unexpected event type "${type}". Expected "message".`);
+      }
+      listeners.splice(listeners.findIndex((l) => l === listener));
+    },
+    postMessage: (data) => {
+      if (!ref.current) {
+        if ("value" in data && data.value === SYMBOL_IGNORING_RPC_RESPONSE_ERROR) {
+          return;
+        }
+        throw Error("Failed to return RPC response to WebView via postMessage");
+      }
+      console.debug("[webview:res]", data);
+      const dataStr = JSON.stringify(data);
```

Comment:
Or handler for bigint values might be enough

Reply:
it's expected to send and receive JSON RPC compliant data, so bigints get encoded as strings.

if you use a library like wagmi it will handle this for you, you pass in bigints and it'll take care of serializing and deserializing